src/lib/arch/riscv/Instruction_decode.cc

```diff
@@ -205,9 +205,15 @@ void Instruction::decode() {
     else if (i > 0 && op.type == RISCV_OP_MEM) {
       //  Memory operand
       sourceRegisters[sourceRegisterCount] = csRegToRegister(op.mem.base);
+      imm_ = op.mem.disp;
       sourceRegisterCount++;
       operandsPending++;
     }
+
+    // First operand is never immediate
```

Comment:
I'm only slightly confident on this but I think it would be better to move the comments above the `else if` into the body of the conditional to align with the rest of the codebase